src/core/TxAtomicSimple.sol

```diff
@@ -50,19 +257,169 @@ abstract contract TxAtomicSimple is IBCKeeper, PacketHandler, ContractRegistry,
         return packPacketAcknowledgementCall(ack);
     }
 
-    function handleAcknowledgement(
-        Packet memory,
-        /*packet*/
-        bytes memory /*acknowledgement*/
-    )
-        internal
-        virtual
-        override
-    {
-        revert NotImplemented();
+    /**
+     * @dev Coordinator side: Handle ACK, update CoordinatorState, and execute Commit/Abort.
+     */
+    function _handleAcknowledgement(Packet calldata packet, bytes calldata acknowledgement) internal virtual override {
+        // --- 1. Decode Acknowledgement ---
+
+        Acknowledgement.Data memory ackOuter = Acknowledgement.decode(acknowledgement);
+        // Simple protocol assumes is_success=true
+        if (!ackOuter.is_success) {
+            revert AckIsNotSuccess();
+        }
+
+        PacketData.Data memory ackPd = PacketData.decode(ackOuter.result);
+        if (ackPd.payload.length == 0) revert PayloadDecodeFailed();
+
+        Any.Data memory ackAny = Any.decode(ackPd.payload);
+        // solhint-disable-next-line gas-small-strings
+        if (sha256(bytes(ackAny.type_url)) != sha256(bytes("/cross.core.atomic.simple.PacketAcknowledgementCall"))) {
+            revert UnexpectedTypeURL();
+        }
+
+        PacketAcknowledgementCall.Data memory ack = PacketAcknowledgementCall.decode(ackAny.value);
+
+        // --- 2. Recover txID from original Packet ---
+
+        PacketData.Data memory callPd = PacketData.decode(packet.data);
+        if (callPd.payload.length == 0) revert PayloadDecodeFailed();
+
+        Any.Data memory callAny = Any.decode(callPd.payload);
+        // solhint-disable-next-line gas-small-strings
+        if (sha256(bytes(callAny.type_url)) != sha256(bytes("/cross.core.atomic.simple.PacketDataCall"))) {
+            revert UnexpectedTypeURL();
+        }
+        PacketDataCall.Data memory pdc = PacketDataCall.decode(callAny.value);
+
+        if (pdc.tx_id.length != 32) {
+            revert InvalidTxIDLength();
+        }
+        bytes32 txID = abi.decode(pdc.tx_id, (bytes32));
+
+        // --- 3. Retrieve & Validate CoordinatorState ---
+
+        CoordStorage storage coordStorage = _getCoordStorage();
+        TxStorage storage txStorage = _getTxStorage();
+
+        CoordEntry storage entry = coordStorage.states[txID];
+        if (!entry.exists) {
```

Comment:
Instead of `exists`, wouldn’t it be sufficient to check whether `commit_protocol` is zero?